src/v/resource_mgmt/storage.cc

```diff
@@ -643,4 +664,94 @@ ss::future<> disk_space_manager::manage_data_disk(uint64_t target_size) {
 }
 
+disk_space_manager::probe::probe(disk_space_manager* sm)
+  : _sm(sm) {}
+
+void disk_space_manager::probe::setup_metrics() {
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    namespace sm = ss::metrics;
+
+    const auto group_name = prometheus_sanitize::metrics_name(
+      "space_management");
+
+    std::vector<sm::impl::metric_definition_impl> defs;
+
+    /*
+     * The currently configured target size for the data disk. When disk
+     * usage exceeds this amount then the expectation is that space
+     * management will reclaim data.
+     *
+     * If space management is disabled the metric value is 0.
+     */
+    defs.emplace_back(sm::make_gauge(
+      "target_disk_size_bytes",
+      [this]() { return _sm->enabled() ? _sm->_target_size : 0; },
+      sm::description("Target maximum number of stored bytes.")));
+
+    defs.emplace_back(sm::make_gauge(
+      "disk_usage_bytes",
+      [this]() { return _total_usage; },
+      sm::description(
+        "Total amount of disk usage under control of space management.")));
+
+    defs.emplace_back(sm::make_gauge(
+      "retention_reclaimable_bytes",
+      [this]() { return _retention_reclaimable; },
+      sm::description("Total amount of reclaimable data through standard "
+                      "retention policy (ref: retention.{ms,bytes}).")));
+
+    defs.emplace_back(sm::make_gauge(
+      "available_reclaimable_bytes",
+      [this]() { return _available_reclaimable; },
+      sm::description("Total amount of available reclaimable data by space "
+                      "management.")));
+
+    defs.emplace_back(sm::make_gauge(
+      "local_retention_reclaimable_bytes",
+      [this]() { return _local_retention_reclaimable; },
+      sm::description(
+        "Total amount of reclaimable data above the local "
+        "retention target (ref: retention.local.target.{ms,bytes}).")));
+
+    defs.emplace_back(sm::make_gauge(
+      "target_excess_bytes",
+      [this]() { return _target_excess; },
+      sm::description("Amount of data usage that exceeds target threshold.")));
+
+    defs.emplace_back(sm::make_gauge(
+      "reclaim_local_bytes",
+      [this]() { return _reclaim_local; },
+      sm::description("Estimated amount of data above local retention to be "
+                      "reclaimed by space management")));
+
+    defs.emplace_back(sm::make_gauge(
+      "reclaim_low_non_hinted_bytes",
+      [this]() { return _reclaim_low_non_hinted; },
+      sm::description("Estimated amount of data above the non-hinted low-space "
+                      "threshold to be reclaimed by space management")));
+
+    defs.emplace_back(sm::make_gauge(
+      "reclaim_low_hinted_bytes",
+      [this]() { return _reclaim_low_hinted; },
+      sm::description("Estimated amount of data above the hinted low-space "
+                      "threshold to be reclaimed by space management")));
```

Comment:
nit: i don't feel too strongly, but just making sure it's intentional: i often think about "older data" being "lower", so I'd lean towards calling this "below the hinted threshold"

Reply:
Hmm, yeh good point. IIRC the documentation we wrote has the data pictured vertically where data "above" is older (the active segment is at the bottom). Aligning with docs seems like the right thing to do?

Reply:
👍  SGTM. For my own reference, are you referring to public docs Or an internal diagram?

I see https://docs.redpanda.com/current/manage/cluster-maintenance/disk-utilization/ but from what I can tell that's kind of illustrating new data being above and to the right

Reply:
Hmm, yeh the picture there seems to be new, but it's also confusing. THe active segment is at the bottom, but the time arrow points up as if new data arrives at the top instead of the active segment. this is the picture from design docs and i think was at some point similar to the public docs

<img width="650" alt="image" src="https://github.com/redpanda-data/redpanda/assets/242417/ec433dcd-b457-430f-937c-e16e1f19e1b7">